### PR TITLE
Fixing minor SpendingByAward bugs

### PR DESCRIPTION
**Pull request by tony-sappe**
**Description:**
Some request edge cases were not being handled properly by the API logic.

**Technical details:**
When "no intersection" is provided in `award_type_codes` then there will never be results returned. So it is better to explicitly return 0 results early in the API logic rather than let the request processing continue to allow exceptions to occur in this edge case.

**Requirements for PR merge:**

1. [x] Unit & integration tests updated
2. [x] API documentation updated (N/A)
3. [ ] Necessary PR reviewers:
	- [ ] Backend
4. [x] Matview impact assessment completed (N/A)
5. [ ] Frontend impact assessment completed
6. [x] Data validation completed (N/A)
7. [x] Appropriate Operations ticket(s) created (N/A)
8. [x] Jira Ticket [DEV-1560](https://federal-spending-transparency.atlassian.net/browse/DEV-1560):
	- [x] Link to this Pull-Request
	- [x] Performance evaluation of affected API
	- [x] Before / After data comparison

**Area for explaining above N/A when needed:**
```
API documentation updated: No edits to API contract, just better enforcement of existing rules
Matview impact assessment completed: No matview edits
Data validation completed: No data edits or data logic edits
Appropriate Operations ticket(s) created: No data changes
```

### CODE REVIEW

File: usaspending_api/search/v2/views/search.py [hunk 1]
```diff
@@ -277,8 +277,8 @@ def post(self, request):
                 m['optional'] = False
 
         json_request = TinyShield(models).block(request.data)
-        fields = json_request.get("fields", None)
-        filters = json_request.get("filters", None)
+        fields = json_request["fields"]
+        filters = json_request.get("filters", {})
```
Comment (dpb-bah) [inline usaspending_api/search/v2/views/search.py]: Should this logic also apply to SpendingByAwardCount (line 434)

File: usaspending_api/search/v2/views/search.py [hunk 2]
```diff
@@ -287,6 +287,14 @@ def post(self, request):
         lower_limit = (page - 1) * limit
         upper_limit = page * limit
 
+        if "no intersection" in filters["award_type_codes"]:
```
Comment (dpb-bah) [inline usaspending_api/search/v2/views/search.py]: Should this check also be applied to `SpendingByAwardCount`?

File: usaspending_api/search/v2/views/search.py (hunk 1 above)
Comment (tony-sappe) [reply usaspending_api/search/v2/views/search.py]: I thought about this, but I didn't want to make those changes without testing. It would also be necessary for SpendingByTransactionCount

Comment (dpb-bah) [reply usaspending_api/search/v2/views/search.py]: The SpendingByTransaction count point I agree would be more. I just bring this up considering SpendingByAward and SpendingByAwardCount go hand-in-hand and their numbers should match.

Comment (tony-sappe) [reply usaspending_api/search/v2/views/search.py]: I'll add it to SpendingByAwardCount since I have time.
I checked SpendingByTransactionCount and it doesn't support that filter so it is a moot point
